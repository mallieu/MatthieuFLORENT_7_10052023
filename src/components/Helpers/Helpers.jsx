import PropTypes from 'prop-types';

// Fonction permettant de parcourir une liste et de générer des éléments
// Nécessite data en array
// Génération par défaut sous forme de liste
// Peut produire des div ou des composants
// /!\ Si isComponent = true, component est défini et isDiv = true
// Ex : CollapseModule

const MapList = ({
    data,
    container_Class,
    component_Class,
    hasBackgroundImage = false,
    isDiv = false,
    isComponent = false,
    component: Component, // Destructuration + renommage du composant
}) => {
    const Container = isComponent ? 'div' : 'ul';
    const Item = isDiv ? 'div' : 'li';

    return (
        <Container className={container_Class}>
            {data.map((item, index) => {
                const { id, cover } = item;
                let key = id ? id : item;

                if (typeof key === 'object') {
                    const values = Object.values(item);
                    key = values[0];
                }

                const itemStyle = hasBackgroundImage
                    ? { backgroundImage: `url(${cover})` }
                    : null;

                return (
                    <Item
                        className={component_Class}
                        key={key}
                        style={itemStyle}
                    >
                        {isComponent ? (
                            <Component data={item} index={index} />
                        ) : (
                            item
                        )}
                    </Item>
                );
            })}
        </Container>
    );
};

MapList.propTypes = {
    data: PropTypes.array.isRequired,
    container_Class: PropTypes.string,
    component_Class: PropTypes.string,
    hasBackgroundImage: PropTypes.bool,
    isDiv: PropTypes.bool,
    isComponent: PropTypes.bool,
    component: PropTypes.elementType,
};

export { MapList };
