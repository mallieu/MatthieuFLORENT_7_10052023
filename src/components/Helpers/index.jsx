const MapList = ({ data, component, component_Class, ul_Class }) => {
    return (
        <ul className={ul_Class}>
            {data.map((item) => (
                <li className={component_Class} key={item}>
                    {item}
                </li>
            ))}
        </ul>
    );
};

export { MapList };
