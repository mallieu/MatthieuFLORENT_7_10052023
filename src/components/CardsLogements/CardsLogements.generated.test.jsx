import renderer from 'react-test-renderer';
import useAPICall from '../../assets/API/API';
import { MapList } from '../../components/Helpers/Helpers';
import ListeAnnonces from './CardsLogements';

jest.mock('../../assets/API/API');
jest.mock('../../components/Helpers/Helpers');

const renderTree = (tree) => renderer.create(tree);
describe('<ListeAnnonces>', () => {
    it('ListeAnnonces - should render component <ListAnnonces />', () => {
        expect(renderTree(<ListeAnnonces />).toJSON()).toMatchSnapshot();
    });
});
