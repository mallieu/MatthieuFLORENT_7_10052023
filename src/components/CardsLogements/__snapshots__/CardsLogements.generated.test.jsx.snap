// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListeAnnonces> ListeAnnonces - should render component <ListAnnonces /> 1`] = `null`;
